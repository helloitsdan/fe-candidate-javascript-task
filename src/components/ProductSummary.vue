<template>
  <div>
    <h3>{{ type }} product</h3>

    <p class="is-marginless">
      <strong>{{ price.monthly | formatCurrency }}</strong>
      monthly /
      <strong>{{ price.annual | formatCurrency }}</strong>
      annualy
    </p>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "ProductSummary",
  props: {
    type: String,
    price: Object
  }
});
</script>
