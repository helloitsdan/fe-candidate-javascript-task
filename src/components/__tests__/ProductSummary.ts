import { shallowMount } from "@vue/test-utils";

import ProductSummary from "../ProductSummary.vue";

describe("ProductSummary", () => {
  const fakeProduct = {
    type: "BASIC",
    price: {
      monthly: 100.0,
      annual: 1200.0
    }
  };

  it("renders", () => {
    const wrapper = shallowMount(ProductSummary, {
      propsData: fakeProduct
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
