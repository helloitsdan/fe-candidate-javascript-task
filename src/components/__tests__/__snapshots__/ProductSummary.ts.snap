// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSummary renders 1`] = `
<div>
  <h3>
    BASIC product
  </h3>
   
  <p
    class="is-marginless"
  >
    <strong>
      £100.00
    </strong>
    
    monthly /
    
    <strong>
      £1200.00
    </strong>
    
    annualy
  
  </p>
</div>
`;
